sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/base_execution_context.py

```diff
@@ -136,12 +137,24 @@ async def _fetch_items_helper_with_retries(self, fetch_function):
         # ExecuteAsync passes retry context parameters (timeout, operation start time, logger, etc.)
         # The callback need to accept these parameters even if unused
         # Removing **kwargs results in a TypeError when ExecuteAsync tries to pass these parameters
-        async def callback(**kwargs):  # pylint: disable=unused-argument
-            return await self._fetch_items_helper_no_retries(fetch_function)
-
-        return await _retry_utility_async.ExecuteAsync(
-            self._client, self._client._global_endpoint_manager, callback, **self._options
-        )
+        async def execute_fetch():
+            async def callback(**kwargs):  # pylint: disable=unused-argument
+                return await self._fetch_items_helper_no_retries(fetch_function)
+
+            return await _retry_utility_async.ExecuteAsync(
+                self._client, self._client._global_endpoint_manager, callback, **self._options
+            )
+
+        while True:
+            try:
+                return await execute_fetch()
```

Comment:
I am curious, how will this loop break in case of no exceptions?

Reply:
the return statement breaks the loop if it works